apps/internal/base/internal/storage/partitioned_storage.go

```diff
@@ -148,13 +159,20 @@ func (m *PartitionedManager) Write(authParameters authority.AuthParams, tokenRes
 		localAccountID := idTokenJwt.LocalAccountID()
 		authorityType := authParameters.AuthorityInfo.AuthorityType
 
+		var preferredUsername string
+		if idTokenJwt.PreferredUsername != "" {
+			preferredUsername = idTokenJwt.PreferredUsername
+		} else {
+			preferredUsername = idTokenJwt.UPN
+		}
```

Comment:
Since there's only one condition, you can simply initialize with the value the `else` branch would assign:
```suggestion
		preferredUsername := idTokenJwt.UPN
		if idTokenJwt.PreferredUsername != "" {
			preferredUsername = idTokenJwt.PreferredUsername
		}
```